src/wallet/wallet.cpp

```diff
@@ -2260,22 +2260,36 @@ bool CWallet::DisplayAddress(const CTxDestination& dest)
     return signer_spk_man->DisplayAddress(scriptPubKey, signer);
 }
 
-void CWallet::LockCoin(const COutPoint& output)
+bool CWallet::LockCoin(const COutPoint& output, WalletBatch* batch)
 {
     AssertLockHeld(cs_wallet);
     setLockedCoins.insert(output);
```

Comment:
Re https://github.com/bitcoin/bitcoin/pull/23065#issuecomment-926508651

> > lock(utxo, persistent=false) but utxo already has persisted lock - should delete from db?
> 
> I don't think we should allow re-locking in any case, better to require explicitly unlocking then locking again.

If you want to implement this breaking change then use the return value of `insert()` to know if the output was already locked and return false.

Otherwise, the lock can be persisted but not in memory.

Reply:
> Otherwise, the lock can be persisted but not in memory.

Sorry, I'm not sure how this could happen. Upon thinking further, I think its fine to upgrade a memory-only lock to a persistent lock (this is useful if fundrawtransaction locked the spends and we want to persistently lock them afterward). But how could we end up with a persistent lock not in memory?

Reply:
> But how could we end up with a persistent lock not in memory?

Right, doesn't happen now that unlock clears from memory and db.

Reply:
But does it make sense to allow memory-only lock if it is already persisted?